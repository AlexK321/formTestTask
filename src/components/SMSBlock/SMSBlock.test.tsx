// import React from 'react';
// import { shallow, ShallowWrapper } from 'enzyme';
// import axios from 'axios';
// import SMSBlock from './SMSBlock';
// import { FormData } from '../../Types/interfaces';
// import SubmitResult from '../SubmitResult/SubmitResult';
// //import SubmitResult from '../SubmitResult/SubmitResult';
// //import { CORRECT_SMS_CODE } from '../Form/constants';

// interface ParamTypes {
//   setSMSBlock: (arg0: boolean) => void;
//   formDataValues?: FormData;
// }

// jest.mock('axios');

// describe('Component: SMSBlock', () => {
//   const testProps: ParamTypes = {
//     setSMSBlock: jest.fn(),
//     formDataValues: {
//       userName: 'test',
//       phoneNumber: 'test',
//       email: 'test',
//     },
//   };

//   const setUp = (props?: ParamTypes) => shallow(<SMSBlock {...props} />);
//   let component: ShallowWrapper;

//   beforeEach(() => {
//     component = setUp(testProps);
//   });

//   it('match snapshot component', () => {
//     expect(component).toMatchSnapshot();
//   });

//   it('should call setSMSBlock', () => {
//     const button = component.find('Button');

//     button.simulate('click');
//     expect(testProps.setSMSBlock).toHaveBeenCalledTimes(1);
//   });

//   // describe('when call onSearch', () => {
//   //   beforeEach(() => {
//   //     const search = component.find('Search');

//   //     search.simulate('search');
//   //   });

//   //   describe('when form is finished with result ', () => {
//   //     beforeEach(() => {
//   //       const search = component.find('Search');

//   //       search.simulate('search');
//   //     });

//   //     it('should render SubmitResult', () => {
//   //       console.log(component.debug());
//   //       expect(component.find(SubmitResult)).toHaveLength(1);
//   //     });
//   //   });
//   // });

//   // test onSearch
//   describe('when request status exists', () => {
//     beforeEach(() => {
//       (axios.post as jest.Mock).mockResolvedValue({ data: { smsCode: '1111' } });
//       component = setUp(testProps);
//     });

//     describe('when form is finished with result ', () => {
//       beforeEach(() => {
//         const search = component.find('Search');

//         search.simulate('search');
//       });

//       it('should render SubmitResult', () => {
//         console.log(component.debug());
//         expect(component.find(SubmitResult)).toHaveLength(1);
//       });
//     });
//   });

//   xdescribe('when request ///', () => {
//     beforeEach(() => {
//       (axios.post as jest.Mock).mockRejectedValue('test error');
//     });
//     component = setUp(testProps);
//     const search = component.find('Search');

//     describe('when form is finished with result ', () => {
//       beforeEach(() => {
//         search.simulate('search');
//       });

//       it('should render SubmitResult', () => {
//         expect(component).toEqual({});
//       });
//     });
//   });

//   describe('when request status exists', () => {
//     beforeEach(() => {
//       (axios.post as jest.Mock).mockResolvedValue({ data: { smsCode: '1112' } });
//       component = setUp(testProps);
//     });

//     describe('when form is finished with result ', () => {
//       beforeEach(() => {
//         const search = component.find('Search');

//         search.simulate('search');
//         search.simulate('search');
//         //search.simulate('search');
//       });

//       it('should render SubmitResult', () => {
//         // const search = component.find('Search');

//         // search.simulate('search');
//         // search.simulate('search');
//         // search.simulate('search');

//         const description = component.find('.sms-warning');

//         console.log(description.debug());
//         console.log(description.children().debug());
//         //expect(1).toEqual(1);
//         expect(description.text()).toEqual('Неправильный код. Осталось попытки.');
//       });
//     });
//   });
// });
