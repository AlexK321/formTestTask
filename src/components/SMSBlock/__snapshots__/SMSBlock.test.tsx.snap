// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: SMSBlock match snapshot component 1`] = `
<div
  className="form-row sms-field"
>
  <div>
    <Search
      enterButton=">"
      onSearch={[Function]}
      placeholder="SMS-код"
      size="large"
    />
  </div>
  <Button
    onClick={[Function]}
    type="link"
  >
    Отправить еще раз или изменить номер
  </Button>
</div>
`;
